### ISSUE TITLE
Add support for custom headers

### ISSUE THREAD

**Issue by ofhouse** [dealmore/terraform-aws-next-js#9]
See: https://nextjs.org/docs/api-reference/next.config.js/headers

**Comment by benkraus** [comment]
Yeah, I'm needing this to change the header for the `apple-app-site-association` file (it needs to have a content type of application/json). Anyone know of any other workaround for now?

**Comment by ofhouse** [comment]
@benkraus We currently calculate the content type of static files with the help of the [mime package](https://github.com/broofa/mime).
Since `apple-app-site-association` has no file extension the lookup fails to detect the correct mime type here.

As a workaround you could use a rewrite rule instead:

1. Rename the file `public/apple-app-site-association -> public/apple-app-site-association.json` (Content type should then be set correctly by mime package)
2. Add a rewrite rule to your `next.config.js`:
    ```diff
    module.exports = {
       ...
    +  async rewrites() {
    +    return [
    +      {
    +         source: '/apple-app-site-association',
    +         destination: '/apple-app-site-association.json',
    +      },
    +    ]
    +  }
    }
     ```

**Comment by benkraus** [comment]
Nice, that worked great, thanks!

**Comment by crabdul** [comment]
Hey @ofhouse 👋,

Is there anyway our team at https://github.com/octoenergy can assist pushing this forward?
Would be fantastic to be able to use this feature

**Comment by benkraus** [comment]
I lied - I had an old cached version that was rendering when deployed, but deploying to a clean environment yields in nothing being found at the `source`, and only something at the `destination` path. It works just fine locally, but only when deployed through this module is it not working. Could it be that rewrites aren't supported for static files here?

**Comment by benkraus** [comment]
I've been doing more debugging - this might be a separate issue really, but it seems rewrites work just fine to pages when deployed with this module, or with static files when developing locally, however static files deployed with a rewrite rule don't have any effect when deployed through this module.

**Comment by ofhouse** [comment]
Sorry for the delay here 🙈

@benkraus Tested it today with [this example](https://github.com/ofhouse/repro-terraform-aws-next-js-static-rewrite) and the rewrite is also applied to the static file (See https://d1nxi0kwcwado6.cloudfront.net/apple-app-site-association).

**Comment by ofhouse** [comment]
@crabdul Yep the feature is still on the roadmap but not prioritized or planned at this time.
Need some time to create a proof of concept for this, because I am not sure if it would work the way I described it in the first comment.

I generally plan to create a new release each month (`0.8.0`, `0.9.0`, etc.).
The scope for the upcoming [`0.9.0`](https://github.com/dealmore/terraform-aws-next-js/milestone/3) release in May is already frozen.
I can try to prioritize this for the `0.10.0` release which would be due in June.

**Comment by benkraus** [comment]
Absolutely bizarre, thanks for the sample project @ofhouse, I should have done that! Apparently something is just really messed up on my end, but I can see it certainly is working fine here... Appreciate everything!
